torchgeo/trainers/classification.py

```diff
@@ -179,6 +180,19 @@ def validation_step(  # type: ignore[override]
         self.log("val_loss", loss, on_step=False, on_epoch=True)
         self.val_metrics(y_hat_hard, y)
 
+        if batch_idx < 10:
+            try:
+                datamodule = self.trainer.datamodule  # type: ignore[attr-defined]
+                batch["prediction"] = y_hat_hard
+                sample = unbind_samples(batch)[0]
+                fig = datamodule.plot(sample)
```

Comment:
This fails when running train.py because sample is on the GPU so .numpy() does not work

Reply:
Should we move to the CPU in the trainer code or in the dataset plot functions? I'm leaning towards the latter so that the code always works.

Reply:
I would say trainer. If you try to call .cpu().numpy() on something that has requires_grad then it will throw an error (saying you need to detach first) which would be a confusing thing to happen in a plot function.

Reply:
But if we do it in the trainer, then any user not using the trainer will still hit this error and have to manually move to the CPU...

Reply:
Sure, but I think it is fine for plot methods to expect inputs to be on the CPU. Another way of looking at it: as a user I would think it was weird if I accidentally called plot with a GPU tensor and as a side effect it moved that tensor to the CPU.